README.md

```diff
@@ -2,42 +2,339 @@
+
+```julia
+using CUDA
+CUDA.allowscalar(false)
+
+using GeoParams, GLMakie, CellArrays
+using JustRelax, JustRelax.DataIO
+```
+Setup of environment and general solution for the benchmark test.
+
+```julia
+# setup ParallelStencil.jl environment
+model  = PS_Setup(:CUDA, Float64, 2)
+environment!(model)
+
+# HELPER FUNCTIONS ---------------------------------------------------------------
+solution(ε, t, G, η) = 2 * ε * η * (1 - exp(-G * t / η))
+
+```
+The function `init_phases!` initializes the phases within cell arrays. The function is parallelized with `@parallel_indices (i,j) ` and `@parallel (JustRelax.@idx ni)`. In this case, this is the only function that needs to be taylored to the specific problem, everything else is handled by `JustRelax.jl` itself. 
+
+```julia
+
+# Initialize phases on the particles
+function init_phases!(phase_ratios, xci, radius)
+    ni      = size(phase_ratios.center)
+    origin  = 0.5, 0.5
+    
+    @parallel_indices (i, j) function init_phases!(phases, xc, yc, o_x, o_y)
+        x, y = xc[i], yc[j]
+        if ((x-o_x)^2 + (y-o_y)^2) > radius^2
+            JustRelax.@cell phases[1, i, j] = 1.0
+            JustRelax.@cell phases[2, i, j] = 0.0
+        
+        else
+            JustRelax.@cell phases[1, i, j] = 0.0
+            JustRelax.@cell phases[2, i, j] = 1.0
+        end
+        return nothing
+    end
+
+    @parallel (JustRelax.@idx ni) init_phases!(phase_ratios.center, xci..., origin...)
+end
+```
+Initialisation of the physical domain and the grid. As `JustRelax.jl` is utilising [ImplicitGlobalGrid.jl](https://github.com/omlins/ImplicitGlobalGrid.jl), the grid can be `MPIAWARE` through setting the grid steps in x- and y- direction to ` di = @. li / (nx_g(),ny_g())`. This makes it a global grid and the grid steps are automatically distributed over the MPI processes. 
+
+```julia
+
+# MAIN SCRIPT --------------------------------------------------------------------
+function main(igg; nx=64, ny=64, figdir="model_figs")
+
+    # Physical domain ------------------------------------
+    ly       = 1e0          # domain length in y
+    lx       = ly           # domain length in x
+    ni       = nx, ny       # number of cells
+    li       = lx, ly       # domain length in x- and y-
+    di       = @. li / ni   # grid step in x- and -y
+    origin   = 0.0, 0.0     # origin coordinates
+    xci, xvi = lazy_grid(di, li, ni; origin=origin) # nodes at the center and vertices of the cells
+    dt       = Inf 
+```
+Initialisation of the Rheology utilising [GeoParams.jl](https://github.com/JuliaGeodynamics/GeoParams.jl). The Rheology can be taylored to the specific problem with different creep laws and material parameters (see [GeoParams.jl](https://github.com/JuliaGeodynamics/GeoParams.jl)) or the miniapps in the [convection folder](miniapps/convection).
```

Comment:
~~Initialisation of the Rheology utilising GeoParams.jl~~ "Initialisation of the rheology with GeoParams.jl"

Note that `rheology` is no capitalized mid-sentence.